File: split-0s-1s/js/lib/main.js
const segregate = (arr) => {
  let left = 0
  let right = arr.length - 1

  while (left < right) {
    if (arr[left] === 0) {
      left += 1
    }
    if (arr[right] === 1) {
      right -= 1
    }
    if (left < right) {
      let tmp = arr[left]
      arr[left] = arr[right]
      arr[right] = tmp
    }
  }
  return arr
}

module.exports = segregate

File: split-0s-1s/ruby/lib/main.rb
def segregate(arr)
  left, right = 0, arr.length - 1

  while left < right
    left += 1 if arr[left] == 0
    right -= 1 if arr[right] == 1
    if left < right
      arr[left], arr[right] = arr[right], arr[left]
    end
  end

  arr
end
